engines/python/setup/djl_python/huggingface.py

```diff
@@ -496,41 +496,96 @@ def register_adapter(inputs: Input):
+
+    result = _service.rolling_batch.add_lora(adapter_name, adapter_path)
+    if adapter_pin:
+        try:
+            if not result:
+                return Output().error(
+                    "error",
+                    message=f"Failed to register LoRA adapter {adapter_name}")
+            result = _service.rolling_batch.pin_lora(adapter_name)
+            if not result:
+                return Output().error(
+                    "error",
+                    message=f"Failed to register LoRA adapter {adapter_name}")
+        except (ValueError, RuntimeError) as e:
+            if any(msg in str(e)
+                   for msg in ("No free lora slots",
+                               "greater than the number of GPU LoRA slots")):
+                raise MemoryError(str(e))
+            return Output().error(
+                f"Failed to register LoRA adapter {adapter_name}",
+                message=str(e))
+
+    logging.info(
+        f"Registered adapter {adapter_name} from {adapter_path} successfully")
+    return Output(message=f"Adapter {adapter_name} registered")
+
+
+def update_adapter(inputs: Input):
+    """
+    Updates lora adapter with the model.
+    """
     if not is_rolling_batch_enabled(_service.hf_configs.rolling_batch):
-        if isinstance(_service.model, PeftModel):
-            _service.model.load_adapter(adapter_path, adapter_name)
-        else:
-            _service.model = PeftModel.from_pretrained(_service.model,
-                                                       adapter_path,
-                                                       adapter_name)
+        raise NotImplementedError(
+            "LoRA adapter API is only supported for rolling batch.")
 
-        if isinstance(_service.hf_pipeline, Pipeline):
-            _service.hf_pipeline.model = _service.model
+    adapter_name = inputs.get_property("name")
+    adapter_path = inputs.get_property("src")
+    adapter_pin = inputs.get_property("pin") == "true"
+    if adapter_name not in _service.adapter_registry:
+        raise ValueError(f"Adapter {adapter_name} not registered.")
+    _service.adapter_registry[adapter_name] = inputs
 
-        if isinstance(_service.hf_pipeline_unwrapped, Pipeline):
-            _service.hf_pipeline_unwrapped.model = _service.model
-    return Output()
+    if adapter_pin:
+        try:
+            result = _service.rolling_batch.pin_lora(adapter_name)
+            if not result:
+                return Output().error(
+                    "error",
+                    message=f"Failed to pin LoRA adapter {adapter_name}")
+        except (ValueError, RuntimeError) as e:
+            if any(msg in str(e)
+                   for msg in ("No free lora slots",
+                               "greater than the number of GPU LoRA slots")):
+                raise MemoryError(str(e))
+            return Output().error(f"Failed to pin LoRA adapter {adapter_name}",
+                                  message=str(e))
+
```

Comment:
let's add an unsupported operation case for when an update is requesting a currently pinned adapter to be unpinned since we don't support that at the engine level